packages/contracts-bedrock/src/libraries/Compression.sol

```diff
@@ -0,0 +1,83 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.15;
+
+/// @title Compression
+/// @notice Compression is a library containing compression utilities.
+library Compression {
+    /// @notice Version of https://github.com/Vectorized/solady/blob/main/src/utils/LibZip.sol
+    ///         that only returns the length of the data if it were to be compressed. This saves
+    ///         gas over actually compressing the data, given we only need the length.
```

Comment:
Do the gas savings here justify modifying the audited function? Solady's `LibZip` was included in the [Cantina audit](https://github.com/Vectorized/solady/blob/main/audits/cantina-solady-report.pdf), so if we can just pull it in as a dependency rather than vendorizing + modifying it, that'd be great.

Reply:
done 👍